backend/src/Types/Auth.hs

```diff
@@ -0,0 +1,29 @@
+module Types.Auth where
+
+import Data.Aeson (FromJSON)
+import Servant (AuthProtect)
+import Servant.Server.Experimental.Auth (AuthServerData)
+
+data SessionIdCookie = SessionIdCookie -- Type-level tag for the auth scheme
+
+type instance AuthServerData (AuthProtect SessionIdCookie) = Authenticated
+
+newtype UserId = UserId {unUserId :: Text}
```

Comment:
Not sure if you're familiar with this idiom.

`newtype UserId = UserId {unUserId :: Text}` is the same as `newtype UserId = UserId Text` but with the added benefit of creating a function `unUserId :: UserId -> Text` that unwraps the newtype. The unwrapping function is named `un<MyType>` by convention.

This can be somewhat convenient in places where a pattern match unwrap might not be stylistically preferred - and you can see a usage in the Database module here.

Reply:
Was not familiar with it, this is handy ✨